js/interface.js

```diff
@@ -2528,13 +2528,42 @@ var DynamicLists = (function() {
         _this.config.likesDataSourceId = '';
       }
       if (_this.config.social.bookmark && (!_this.config.bookmarkDataSourceId || _this.config.bookmarkDataSourceId === '')) {
-        // Create likes data source
+        // Create bookmarks data source
         bookmarksPromise = Fliplet.DataSources.create({
           name: appName + ' - Bookmarks',
-          bundle: true,
-          organizationId: organizationId // optional
+          bundle: false,
+          organizationId: organizationId, // optional
+          definition: {
+            views: [
+              {
+                name: 'userBookmarks',
+                bundle: true,
+                filter: {
+                  user: {
+                    $or: [
+                      // Guest users
+                      { uuid: 'session.uuid' },
+
+                      // Logged in users
+                      { email: 'session.email' },
+
+                      // Legacy support just in case
+                      { dataSourceEntryId: 'session.id' }
+                    ]
+                  }
+                }
+              }
+            ]
+          }
         }).then(function (dataSource) {
           _this.config.bookmarkDataSourceId = dataSource.id;
+          _this.config.dataSourceViews = _this.config.dataSourceViews || [];
+
+          _this.config.dataSourceViews.push({
+            id: dataSource.id,
+            name: 'userBookmarks',
+            bundle: true
+          })
```

Comment:
Let's run a `_.uniq` through `_this.config.dataSourceViews` to make sure we don't have any duplicated views of the same data source. The mechanism for managing bookmark/like/comment data sources don't exist at the moment, but I'd play it safe just in case?

Reply:
@tonytlwu done, check latest commit.